docs/source/configuration/config_loader_migration.md

```diff
@@ -60,3 +60,177 @@ In this example, `"catalog"` is the key to the default catalog patterns specifie
+
+### 1. Install the required library
+The [`OmegaConfigLoader`](advanced_configuration.md#omegaconfigloader) was introduced in Kedro `0.18.5` and is based on [OmegaConf](https://omegaconf.readthedocs.io/). Features that replace `TemplatedConfigLoader` functionality have been released in later versions, so we recommend users
+to install at least Kedro version `0.18.X` to properly replace the `TemplatedConfigLoader` with `OmegaConfigLoader`.
+You can install both this Kedro version and `omegaconf` using `pip`:
+
+```bash
+pip install kedro==0.18.X
+```
+This would be the minimum required Kedro version which includes `omegaconf` as a dependency and the necessary functionality to replace `TemplatedConfigLoader`.
+Or you can run:
+```bash
+pip install -U kedro
+```
+
+This command installs the most recent version of Kedro which also includes `omegaconf` as a dependency.
+
+### 2. Use the `OmegaConfigLoader`
+To use `OmegaConfigLoader` in your project, set the `CONFIG_LOADER_CLASS` constant in your [`src/<package_name>/settings.py`](../kedro_project_setup/settings.md):
+
+```diff
++ from kedro.config import OmegaConfigLoader  # new import
+
++ CONFIG_LOADER_CLASS = OmegaConfigLoader
+```
+
+### 3. Import statements
+Replace the import statement for `TemplatedConfigLoader` with the one for `OmegaConfigLoader`:
+
+```diff
+- from kedro.config import TemplatedConfigLoader
+
++ from kedro.config import OmegaConfigLoader
+```
+
+### 4. File format support
+`OmegaConfigLoader` supports only `yaml` and `json` file formats. Make sure that all your configuration files are in one of these formats. If you were using other formats with `TemplatedConfigLoader`, convert them to `yaml` or `json`.
+
+### 5. Load configuration
+The method to load the configuration using `OmegaConfigLoader` differs slightly from that used by `TemplatedConfigLoader`, which allowed users to access configuration through the `.get()` method and required patterns as argument.
+When you migrate to use `OmegaConfigLoader` it  requires you to fetch configuration through a configuration key that points to [configuration patterns specified in the loader class](configuration_basics.md#configuration-patterns) or [provided in the `CONFIG_LOADER_ARGS`](advanced_configuration.md#how-to-change-which-configuration-files-are-loaded) in `settings.py`.
+
+```diff
+- conf_path = str(project_path / settings.CONF_SOURCE)
+- conf_loader = TemplatedConfigLoader(conf_source=conf_path, env="local")
+- catalog = conf_loader.get("catalog*")
+
++ conf_path = str(project_path / settings.CONF_SOURCE)
++ config_loader = OmegaConfigLoader(conf_source=conf_path, env="local")
++ catalog = config_loader["catalog"] # note the key accessor syntax
+```
+
+In this example, the `"catalog"` key points to the default catalog patterns specified in the `OmegaConfigLoader` class.
+
+### 6. Templating of values
```

Comment:
I think, under this section, it would also be helpful to flag that the variable interpolation is only scoped to a particular configuration and the same environment. For eg., templated keys from `local` will not overwrite `base` and the resolution of the keys happens within the environment which is different from how `TemplatedConfigLoader` uses the globals.

Reply:
Second on this. Do we already have the relevant docs in OmegaConfigLoader's page? Obviously just move everything to `global` would work(this is how template config loader works), but we may want to advise to use the `catalog_globals.yml` etc to replace the use of `globals.py` for certain things.